### PR TITLE
20240207 invite as org

### LINKED ISSUE #86: `inviteHumanAsOrganization` as an organization
we can have organizations invite people. If the org has no Circles on their account, there is no liquidity for people invited by org initially, but it does set them up in the system; ready to go find more people. (If the org has a balance of tokens, then there is some liquidity)

the motivation for `invited by org` is for people at events, to be able to wear a different hat, and rather than being forced to invite from your personal account, you can onboard people from the organization.

the interface would have a donation in XDAI, which can be arbitrary, but not-zero value, purely as a reputational game to "make it count" to invite people as an org (because who's CRC to burn); later we can build mechanisms to extend this, where the donation should get transferred. (benefit: because it is msg value, we can actively call on the donation address without ERC1155 standard)

`function inviteHumanAsOrganization(address _human, address _donationDestination)`

### PR DESCRIPTION
**Pull request by benjaminbollen**
⚠️ ~~I ll rebase this once the other PR is also on develop~~, thanks @jaensen / ✅  done

### CODE REVIEW

**Comment by benjaminbollen** [comment]
that seems to be now 🎉

File: src/hub/Hub.sol [hunk 1]
```diff
@@ -262,6 +269,41 @@ contract Hub is Circles {
...
 
+    /**
+     * Invite human as organization allows to register a human avatar as an organization.
+     * @param _human address of the human to invite
+     * @param _donationReceiver address of where to send the donation to with 2300 gas (using transfer)
+     */
+    function inviteHumanAsOrganization(address _human, address payable _donationReceiver) external payable {
+        require(msg.value > MINIMUM_DONATION, "Donation must be at least 0.1 xDai.");
+        require(isOrganization(msg.sender), "Only organizations can invite.");
+
```
Comment (jaensen) [inline src/hub/Hub.sol:262]:
```suggestion
require(msg.sender != _donationReceiver, "Can't donate to yourself");
```

Comment (jaensen) [reply src/hub/Hub.sol:262]: Makes it at least a little bit more expensive to perpetually invite people with the same funds.

File: src/hub/Hub.sol [hunk 2]
```diff
@@ -262,6 +269,41 @@ contract Hub is Circles {
...
+        // insert avatar into linked list; reverts if it already exists
+        _insertAvatar(_human);
+
+        // set the last mint time to the current timestamp for invited human
+        // and register the v1 Circles contract status
+        address v1CirclesStatus = _avatarV1CirclesStatus(_human);
```
Comment (jaensen) [inline src/hub/Hub.sol]: Let's refactor the common parts of `inviteHuman` and `inviteHumanAsOrganization` into an _inviteHuman function

**Review by jaensen — CHANGES_REQUESTED** [review]
small refactoring

File: src/hub/Hub.sol (hunk 1 above)
Comment (benjaminbollen) [reply src/hub/Hub.sol:262]: well, I agree, but you can always send to yourself over one-hop, so I don't know if we should block it. we already know that the donation is a reputation requirement for it to have meaning

File: src/hub/Hub.sol (hunk 2 above)
Comment (benjaminbollen) [reply src/hub/Hub.sol]: yes, I wanted to, but then thought I was overthinking it. Ill quickly do

File: src/hub/Hub.sol (hunk 1 above)
Comment (benjaminbollen) [reply src/hub/Hub.sol:262]: I added the requirement with comment

File: src/hub/Hub.sol (hunk 2 above)
Comment (benjaminbollen) [reply src/hub/Hub.sol]: turned out it was well worth the refactoring!

**Review by jaensen — APPROVED** [review]
fipf

**Pull request merged**